src/parsers/pm_io/data.jl

```diff
@@ -1796,3 +1796,28 @@ function _make_multiconductor(data::Dict{String,Any}, conductors::Real)
         end
     end
 end
+
+
+"""
+Move gentype and genfuel fields to be subfields of gen
+"""
+function move_genfuel_and_gentype!(data::Dict{String,Any})
+    ngen = length(data["gen"])
+    
+    toplevkeys = ("genfuel", "gentype")
+    sublevkeys = ("fuel", "type")
+    for i in range(1, stop=2)
+        if haskey(data, toplevkeys[i])
+            # check that lengths of category and generators match
+            if length(data[toplevkeys[i]]) != ngen
+                str = toplevkeys[i]
+                throw("length of $str does not equal the number of generators")
```

Comment:
Can you throw a specific exception?  Would this be a user-input error?

Reply:
Can you point me to documentation or an example for throwing specific exceptions? The error is due to a malformed input file, so I guess "user-input" would fit.

Reply:
Search the codebase for DataFormatError.